README.md

```diff
@@ -36,6 +37,65 @@ to run `git mv old new` on each rename, you can do something like this:
 ls | renamer --rename-command "git mv"
 ```
 
+### Caveat emptor
+
+**NB:** it makes sense to be aware of the issues `ls` can cause in this
+context, depending on the `ls` flavor (or substitute, such as `lsd`, `exa`
+...) used. Please read [this document](https://web.archive.org/web/20230102124738/http://mywiki.wooledge.org/ParsingLs)
+for more information.
+
+While your shell will pass the file names individually, no matter if they
+contain whitespace, an `ls` that fails to detect the pipe and print one file
+name per line will cause issues. Unfortunately `ls -Q` also isn't a solution
+here, because unlike the shell -- which will strip quotes prior to passing
+them to invoked commands -- `renamer` won't handle the quoted names and will
+probably complain about non-existent files, too.
+
+### Advanced usage
+
+If you have tools like GNU `find` at your disposal, you can also use the
+following method:
+
+```bash
+find -type f -exec renamer {} +
+```
+
+This would execute `renamer` with all of the files matched by `find`. You can
+use additional `find` predicates such as `-name` or `-ipath` to limit which
+files to rename. There is, however, one caveat: on large lists of files you
+may encounter multiple invocations of `renamer` -- and thus your editor -- due
+to how `find ... -exec {} +` works. It will pass as many file names on the
+command line as it can fit but it is limited by `ARG_MAX` (see `getconf ARG_MAX`
+output for how long the overall command line length can be on your system).
+
+Other `find` flavors would allow the following, but it would invoke `renamer`
+-- and thus your editor -- *once for every single found file*:
+
+```bash
+find -type f -exec renamer {} \;
+```
+
+In order to sidestep this issue, you can employ `xargs` in conjunction with
+`find` like so (`-print` is implied for `find`):
+
+```
+find -type f|xargs renamer --editor vim
```

Comment:
nit: could you add a space on either side of the pipe here?

Reply:
Done